demo/cloudify_server

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+REPL = true
+
+require_relative "../loader"
+require "timeout"
+
+LOCATIONS = Option::Locations.map { |l| [l.name, "#{l.provider.display_name} #{l.display_name}"] }.to_h
+
+def get_input(msg, default = nil)
+  prompt = "#{msg}#{default.nil? ? "" : " [default: #{default}]"}: "
+  input = ""
+
+  while input.empty?
+    print prompt
+    input = gets.chomp
+    input = default if input.empty? && !default.nil?
+  end
+
+  input
+end
+
+def select_option(msg, options, default = nil)
```

Comment:
Hey Enes, I think this function doesn't properly check for incorrect option index values.

I'm adding an example screenshot below. Could you add basic input checking to this function?

<img width="925" alt="Screenshot 2023-07-31 at 9 41 50 AM" src="https://github.com/ubicloud/ubicloud/assets/2545443/c79e7ac2-a377-4a4e-9aaf-bd0e627a7dd3">

Reply:
Yeah it wasn't working for numbers smaller than 1. I fixed it
```
➜  ubicloud git:(demo-cloudify-helper) ✗ ./demo/cloudify_server
Enter hostname or IP address: 1.1.1.1

1. Hetzner Finland
2. Hetzner Germany
3. DataPacket Istanbul

Select provider and location [default: 1]: 0
Plese enter a number between 1-3 or leave empty for default
Select provider and location [default: 1]: -2
Plese enter a number between 1-3 or leave empty for default
Select provider and location [default: 1]: aa
Plese enter a number between 1-3 or leave empty for default
Select provider and location [default: 1]: bc
Plese enter a number between 1-3 or leave empty for default
Select provider and location [default: 1]: 2


Cloudifying '1.1.1.1' server for 'Hetzner Germany'

```